cli/npm/mod.rs

```diff
@@ -3,422 +3,14 @@
-    if std::env::var("DENO_UNSTABLE_NPM_SYNC_DOWNLOAD") == Ok("1".to_string()) {
-      // for some of the tests, we want downloading of packages
-      // to be deterministic so that the output is always the same
-      let mut packages = self.resolution.all_packages();
-      packages.sort_by(|a, b| a.id.cmp(&b.id));
-      for package in packages {
-        self
-          .cache
-          .ensure_package(&package.id, &package.dist, &self.registry_url)
-          .await
-          .with_context(|| {
-            format!("Failed caching npm package '{}'.", package.id)
-          })?;
-      }
-    } else {
-      let handles = self.resolution.all_packages().into_iter().map(|package| {
-        let cache = self.cache.clone();
-        let registry_url = self.registry_url.clone();
-        tokio::task::spawn(async move {
-          cache
-            .ensure_package(&package.id, &package.dist, &registry_url)
-            .await
-            .with_context(|| {
-              format!("Failed caching npm package '{}'.", package.id)
-            })
-        })
-      });
-      let results = futures::future::join_all(handles).await;
-      for result in results {
-        // surface the first error
-        result??;
-      }
-    }
-    Ok(())
-  }
-
-  fn local_package_info(&self, id: &NpmPackageId) -> LocalNpmPackageInfo {
-    LocalNpmPackageInfo {
-      folder_path: self.cache.package_folder(id, &self.registry_url),
-      id: id.clone(),
-    }
-  }
-
-  /// Creates an inner clone.
-  #[allow(unused)]
-  pub fn snapshot(&self) -> NpmPackageResolverSnapshot {
-    NpmPackageResolverSnapshot {
-      cache: self.cache.as_readonly(),
-      snapshot: self.resolution.snapshot(),
-      registry_url: self.registry_url.clone(),
-    }
-  }
-
-  pub fn get_cache_location(&self) -> PathBuf {
-    self.cache.as_readonly().get_cache_location()
-  }
-}
-
-impl NpmPackageResolver for GlobalNpmPackageResolver {
-  fn resolve_package_from_deno_module(
-    &self,
-    pkg_req: &NpmPackageReq,
-  ) -> Result<LocalNpmPackageInfo, AnyError> {
-    let pkg = self.resolution.resolve_package_from_deno_module(pkg_req)?;
-    Ok(self.local_package_info(&pkg.id))
-  }
-
-  fn resolve_package_from_package(
-    &self,
-    name: &str,
-    referrer: &ModuleSpecifier,
-  ) -> Result<LocalNpmPackageInfo, AnyError> {
-    let referrer_pkg_id = self
-      .cache
-      .resolve_package_id_from_specifier(referrer, &self.registry_url)?;
-    let pkg = self
-      .resolution
-      .resolve_package_from_package(name, &referrer_pkg_id)?;
-    Ok(self.local_package_info(&pkg.id))
-  }
-
-  fn resolve_package_from_specifier(
-    &self,
-    specifier: &ModuleSpecifier,
-  ) -> Result<LocalNpmPackageInfo, AnyError> {
-    let pkg_id = self
-      .cache
-      .resolve_package_id_from_specifier(specifier, &self.registry_url)?;
-    Ok(self.local_package_info(&pkg_id))
-  }
-}
-
-#[derive(Clone, Debug)]
-pub struct NpmPackageResolverSnapshot {
```

Comment:
In the interest of time, I completely removed this for now and will add it back in the future when we actually go to implement lsp support.